Dockerfile

```diff
@@ -0,0 +1,80 @@
+# Heavily inspired by: https://github.com/astral-sh/uv-docker-example/blob/main/multistage.Dockerfile
+
+# Build arguments for flexibility in base image selection
+ARG PYTHON_VERSION=3.13
```

Comment:
Are we sure we want to go with 3.13? Maybe 3.12 would be have more support?

Reply:
what risks do we see with 3.13?

Reply:
not everyone/library publish versions for "everything" so we could encounter that some AIML libraries are not supported yet.

Reply:
Let's leave it up to the template user. With 3.13 used here, we can prove that the common dependencies work with the latest version. If anyone wants to be extra careful or has conflicting dependencies, they can use lower version